test/cmd/diff.sh

```diff
@@ -113,6 +113,89 @@ run_kubectl_diff_tests() {
     # Cleanup
     kubectl delete -f hack/testdata/pod.yaml
     kubectl delete -f hack/testdata/prune/b.yaml
+    kubectl delete namespace nsb
+
+    ## kubectl diff --prune with label selector
+    kubectl create ns nsbprune
+    kubectl apply --namespace nsbprune -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: a
+  namespace: nsbprune
+  labels:
+    prune-group: "true"
+spec:
+  containers:
+  - name: kubernetes-pause
+    image: registry.k8s.io/pause:3.9
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: b
+  namespace: nsbprune
+  labels:
+    prune-group: "true"
+spec:
+  containers:
+    - name: kubernetes-pause
+      image: registry.k8s.io/pause:3.9
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: c
+  namespace: nsbprune
+  labels:
+    prune-group: "false"
+spec:
+  containers:
+    - name: kubernetes-pause
+      image: registry.k8s.io/pause:3.9
+EOF
+    kube::test::get_object_assert 'pods a -n nsbprune' "{{${id_field:?}}}" 'a'
+    kube::test::get_object_assert 'pods b -n nsbprune' "{{${id_field:?}}}" 'b'
+    kube::test::get_object_assert 'pods c -n nsbprune' "{{${id_field:?}}}" 'c'
+    # Make sure that kubectl diff does not return neither pod 'b' nor pod 'c' without prune flag
```

Comment:
```suggestion
    # Make sure that kubectl diff does not return either pod 'b' or pod 'c' without prune flag
```
I think this is more clear.